src/main/java/io/mbab/sda/groupproject/menu/action/CreatePlayerAction.java

```diff
@@ -3,50 +3,102 @@
 import io.mbab.sda.groupproject.entity.City;
 import io.mbab.sda.groupproject.entity.Country;
 import io.mbab.sda.groupproject.entity.Player;
+import io.mbab.sda.groupproject.entity.Team;
 import io.mbab.sda.groupproject.menu.CustomScanner;
 import io.mbab.sda.groupproject.menu.MenuActionContext;
 import io.mbab.sda.groupproject.repository.CityRepository;
 import io.mbab.sda.groupproject.repository.CountryRepository;
+import io.mbab.sda.groupproject.repository.PlayerRepository;
+import io.mbab.sda.groupproject.repository.TeamRepository;
+import lombok.AllArgsConstructor;
 import lombok.RequiredArgsConstructor;
 
 import java.time.LocalDate;
 
+@AllArgsConstructor
 @RequiredArgsConstructor
 public class CreatePlayerAction implements MenuAction {
 
   private final CustomScanner scanner;
   private final MenuActionContext ctx;
   private final CountryRepository countryRepository;
+  private final PlayerRepository playerRepository;
+  private final TeamRepository teamRepository;
 
   @Override
   public void execute() {
 
-    System.out.println("0) Przejdź do poprzedniego menu");
+    Player player;
+
+    System.out.println("!!! DODAJESZ PIŁKARZA !!!");
+    System.out.println("--> Wciśnięcie '0' powoduję powtór do menu głównego <--");
     System.out.println("Podaj imie gracza:");
-    var input = scanner.nextLine();
-    if (pressedZero(input)) return;
-    var builder = Player.builder().firstName(input);
+    String firstName = scanner.nextLine();
+    if (pressedZero(firstName)) return;
 
+    System.out.println("!!! DODAJESZ PIŁKARZA !!!");
     System.out.println("Podaj nazwisko gracza:");
-    input = scanner.nextLine();
-    if (pressedZero(input)) return;
-    var lastName = builder.lastName(input).build();
+    String lastName = scanner.nextLine();
+    if (pressedZero(lastName)) return;
 
+    System.out.println("!!! DODAJESZ PIŁKARZA !!!");
     System.out.println("Podaj date urodzenia gracza:");
-    input = scanner.nextLine();
-    if (pressedZero(input)) return;
-    var date = builder.dateOfBirth(input).build();
+    String dateOfBirth = scanner.nextLine();
+    if (pressedZero(dateOfBirth)) return;
+
+    Team team = null;
+    while (true) {
+      System.out.println("!!! DODAJESZ PIŁKARZA !!!");
+      System.out.println("Podaj drużynę w której gracz występuję:");
+      System.out.println("Pusty znak - piłkarz jest obecnie bez drużyny");
+      String teamName = scanner.nextLine();
+      if (pressedZero(dateOfBirth)) return;
+      if ("".equals(teamName) || " ".equals(teamName)) break;
+      if (teamRepository.findByName(teamName) != null) {
```

Comment:
2 razy wywolujesz findByName, moze lepszym sposobem bedzie jeszeli przeniesiecie to do pomocniczej metody prywatnej i w sytuacji gdy nie istnieje druzyna jeszce rz bedziecie uruchamiac ta metode - rekurencja